api/src/util/newrelic/events.py

```diff
@@ -0,0 +1,74 @@
+import logging
+import sys
+from types import TracebackType
+from typing import Any, Dict, Optional, Tuple, Union
+
+import newrelic.agent
+
+logger = logging.getLogger(__name__)
+
+AGENT_ATTRIBUTE_LIMIT = 128
+EVENT_API_ATTRIBUTE_LIMIT = 255
+
+
+def record_custom_event(event_type: str, params: Dict[Any, Any]) -> None:
+    # legacy aliases
+    params["request.uri"] = params.get("request.path")
+    params["request.headers.x-amzn-requestid"] = params.get("request_id")
+
+    # legacy FineosError attribute aliases
```

Comment:
We can probably adjust some of this / remove Fineos reference.

FYI our logging is a little different now on this project, https://github.com/HHS/simpler-grants-gov/blob/main/api/src/logging/flask_logger.py#L165 has a few more things.

Might be worth adding something like this here since we don't have that x-amzn ID set anywhere yet.
```
if flask.has_request_context():
    params["api.request.internal_id"] = getattr(flask.g, "internal_request_id", None)
```

Reply:
This brings up a good point, especially as we start to look at requests hitting, CDN, LB, backend, S3, DB, OpenSearch, etc. Should we be using the X-amzn one if it's present since it'll get set the earliest in that whole stack? Fall back to our own generated one if we're hitting the API first like locally?

Reply:
Right now, it isn't set. I'm not sure where that values comes from, that value is just what a prior project used, and I never figured out what was setting it in their infra.

Reply:
> We can probably adjust some of this / remove Fineos reference.
> 
> FYI our logging is a little different now on this project, https://github.com/HHS/simpler-grants-gov/blob/main/api/src/logging/flask_logger.py#L165 has a few more things.
> 
> Might be worth adding something like this here since we don't have that x-amzn ID set anywhere yet.
> 
> ```
> if flask.has_request_context():
>     params["api.request.internal_id"] = getattr(flask.g, "internal_request_id", None)
> ```

FYI there is another Fineos reference in api/tool/console/interactive.py.

Reply:
https://kevinhakanson.com/2017-03-08-aws-request-ids/ Looks like it was API Gateway back in the day, but I wonder if CDN might be doing it too now, or if there's another one we should also have in the hierarchy.

Reply:
X-Amz-Cf-Id is the Cloudfront one. So for API code I guess, x-amzn-requestid, falling back to X-Amz-Cf-Id, falling back to our own generated one?
https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/RequestAndResponseBehaviorCustomOrigin.html#request-custom-headers-behavior